### ISSUE TITLE
Dice computation over 1

### ISSUE THREAD

**Issue by Mihai196** [milesial/Pytorch-UNet#56]
Hello. I am trying to use parts of your code with my data set and whenever i evaluate a model  i get values of this sort: 
Epoch finished ! Loss: 0.16545738393068313
Validation Dice Coeff: 1.004147481918335
Checkpoint 1 saved !
Is this normal? Shouldn't dice coefficient be a value between 0 and 1? To be mentioned that my masks are not binary but i load them as binary so the validation set should contain binary images. Do you have any idea why this happens ? 
Thank you a lot for your code and for sharing it. :)

**Comment by milesial** [comment]
If the dice is over 1, I think it's a problem with your masks. Check that the masks' values are between 0 and 1 when they are loaded.

**Comment by Mihai196** [comment]
Well the networks seems to be trained fine and the same code is used for loading images for training and for loading images for validation so there shouldn't be any sort of difference in theory, right?

**Comment by milesial** [comment]
Yes but there is obviously a problem. So just before the dice loss calculation you should check that there is only one channel for the masks, and that all values are in [0, 1].
Moreover, the dice loss is only defined if `n_classes=1`.

**Comment by Mihai196** [comment]
Never mind. The problem was that when i evaluated the final return was total/ index + 1 instead of total / (index+1). I am such a dummy. Sorry for your time and thank you for you instant replies.
